### PR TITLE
fix(curriculum): use regex for background color tests

### LINKED ISSUE #47256: Learn CSS Colors by Building a Set of Colored Markers - Step 49
### Describe the Issue

I believe the code I entered for this particular step is correct but is being parsed as wrong. Maybe I didn't understand what the step entails but I have gone over it several times and input different variations of what I think the correct syntax should be with no success. I have come to the conclusion that it is a bug :)

### Affected Page

https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-css-colors-by-building-a-set-of-colored-markers/step-49

### Your code

```
.red {
  background: rgb(255, 0, 0);
}
```

### Expected behavior

In the .red CSS rule, change the background-color property to background.

### Screenshots

Screen Shot 2022-08-10 at 18 24 06

### System

- Device: [MacBook Air]
- OS: [macOS Monterey]
- Browser: [Safari]
- Version: [15.5]

### Additional context

The same problem occurs in step 60 and step 66
Screen Shot 2022-08-10 at 18 28 19
Screen Shot 2022-08-10 at 18 27 29

### PR DESCRIPTION
**Pull request by lasjorg**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #47256 #46233

Switch to regex because...Safari.

### CODE REVIEW

**Review by majestic-owl448 — CHANGES_REQUESTED** [review]
can you do also the other step? see https://github.com/freeCodeCamp/freeCodeCamp/issues/46233#issuecomment-1141307571

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-colors-by-building-a-set-of-colored-markers/61a489b8579e87364b2d2cdb.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ In the `.red` CSS rule, change the `background-color` property to `background`.
 Your `.red` CSS rule should have a `background` property with the value `rgb(255, 0, 0)`.
 
 ```js
-assert.include(new __helpers.CSSHelp(document).getStyle('.red').cssText, 'background: rgb(255, 0, 0)');
+assert.match(__helpers.removeWhiteSpace(code), /\.red\{.*?background:rgb\(255,0,0\)/);
```
Comment (ieahleen) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-colors-by-building-a-set-of-colored-markers/61a489b8579e87364b2d2cdb.md:23]: could this match the property if written on an other selector?
would it be possible to extract the code for the selector with regex and then check if it includes the right property?

Comment (lasjorg) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-css-colors-by-building-a-set-of-colored-markers/61a489b8579e87364b2d2cdb.md:23]: Not sure I know what you mean, can you give an example? Or if you have a code suggestion I'm open to it.

I should say the same type of regex was already used elsewhere.

Comment (ieahleen) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-css-colors-by-building-a-set-of-colored-markers/61a489b8579e87364b2d2cdb.md:23]: image

I don't think we really have to worry about it, but...

As regex, I would say something like `/\.red{.*?}/` and on the extracted text `.includes("background:rgb(255,0,0)")

Comment (bbsmooth) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-css-colors-by-building-a-set-of-colored-markers/61a489b8579e87364b2d2cdb.md:23]: Just trying to think like a user here. Should probably allow for white space after the commas in (255,0,0\). While we are at it, probably between between the parens and the numbers as well.

Comment (lasjorg) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-css-colors-by-building-a-set-of-colored-markers/61a489b8579e87364b2d2cdb.md:23]: @ieahleen I see what you mean. That seems fairly unlikely to happen unless someone goes out of their way to do something that wasn't asked for.

There are 4 other challenge steps as well that are using that same regex (besides the 3 in this PR).

Comment (ieahleen) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-css-colors-by-building-a-set-of-colored-markers/61a489b8579e87364b2d2cdb.md:23]: It just came to mind, I don't think we have to worry about that too much

**Review by Sembauke — APPROVED** [review]

**Review by majestic-owl448 — APPROVED** [review]

**Pull request merged**